spec/requests/orders_spec.rb

```diff
@@ -0,0 +1,92 @@
+require 'rails_helper'
+
+RSpec.describe "Orders", type: :request do
+  # describe "GET /index" do
+  #   pending "add some examples (or delete) #{__FILE__}"
+  # end
+  describe "get orders_path" do
+    it "renders the index view" do
+      FactoryBot.create_list(:order, 10)
+      get orders_path
+      expect(response).to render_template(:index)
+    end
+  end
+  describe "get order_path" do
+    it "renders the :show template" do
+      order = FactoryBot.create(:order)
+      get order_path(id: order.id)
+      expect(response).to render_template(:show)
+    end
+    it "redirects to the index path if the order id is invalid" do
+      get order_path(id: 5000) #an ID that doesn't exist
+      expect(response).to redirect_to orders_path
+    end
+  end
+  describe "get new_order_path" do
+    it "renders the :new template" do
+      get new_order_path
+      expect(response).to render_template(:new)
+    end
+  end
+  describe "get edit_order_path" do
+    it "renders the :edit template"  do
+      order = FactoryBot.create(:order)
+      get edit_order_path(id: order.id)
+      expect(response).to render_template(:edit)
+    end
+  end
+  describe "post orders_path with valid data" do
+    it "saves a new entry and redirects to the show path for the entry" do
+      customer = FactoryBot.create(:customer)
+      order_attributes = FactoryBot.attributes_for(:order)
+      order_attributes[:customer_id]=customer.id
+      expect { post orders_path, params: {order: order_attributes}
+    }.to change(Order, :count)
+      expect(response).to redirect_to order_path(id: Order.last.id)
+    end
+  end
+  describe "post orders_path with invalid data" do
+    it "does not save a new entry or redirect" do
+      order_attributes = FactoryBot.attributes_for(:order)
+      order_attributes.delete(:first_name)
+      expect { post orders_path, params: {order: order_attributes}
+    }.to_not change(Order, :count)
+      expect(response).to render_template(:new)
+    end
+  end
+  describe "put order_path with valid data" do
+    order = FactoryBot.create(:order)
+    order_attributes = FactoryBot.attributes_for(:order)
+    before do
+      put "/orders/#{order.id}",params: {order: order_attributes}
+    end
+    it "updates an entry and redirects to the show path for the order" do
+      expect(response).to redirect_to order_path(id: Order.last.id)
+    end
```

Comment:
The blocks in which you placed these lines of code are a bit unconventional; I would typically expect this:

```suggestion
    before do
      order = FactoryBot.create(:order)
    end
    it "updates an entry and redirects to the show path for the order" do
      order_attributes = FactoryBot.attributes_for(:order)
      put "/orders/#{order.id}",params: {order: order_attributes}
      expect(response).to redirect_to order_path(id: Order.last.id)
    end
```

That way, if you have more scenarios with different data in your `describe` block, you have a new Order object for each scenario, and you're not forced to make a call with the specific `order_attributes` you defined up top.